js/index.js

```diff
@@ -1,18 +1,35 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
-
-  //... your code goes here
+  
+  const price = product.querySelector('.price span');
+  const quantity = product.querySelector('.quantity input');
+  const priceValue = price.innerHTML;
+  const quantityValue = quantity.value;
+  const subTotalPrice = priceValue * quantityValue;
+  const subTotal = product.querySelector('.subtotal span');
+  const subTotalCalc = subTotal.innerHTML = subTotalPrice;
+
+  return subTotalPrice;
 }
+// looks too long - but apparently works?
 
 function calculateAll() {
   // code in the following two lines is added just for testing purposes.
   // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
-
+  let allProducts = document.querySelectorAll('.product');
+  allProducts = [...allProducts];
+
+  let totalValue = 0;
+  allProducts.forEach(function (element) {
+     updateSubtotal(element);
+     totalValue += updateSubtotal(element)
```

Comment:
line 25 is unnesessary, since we are already getting the result of this directly on line 26.